pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3191,16 +3191,42 @@ func (c *Controller) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineI
 		return vm, common.NewSyncError(fmt.Errorf(fetchingRunStrategyErrFmt, err), FailedCreateReason), err
 	}
 
-	// Ensure we have ControllerRevisions of any instancetype or preferences referenced by the VM
-	err = c.instancetypeMethods.StoreControllerRevisions(vm)
-	if err != nil {
-		log.Log.Object(vm).Infof("Failed to store Instancetype ControllerRevisions for VirtualMachine: %s/%s", vm.Namespace, vm.Name)
-		c.recorder.Eventf(vm, k8score.EventTypeWarning, common.FailedCreateVirtualMachineReason, "Error encountered while storing Instancetype ControllerRevisions: %v", err)
-		return vm, common.NewSyncError(fmt.Errorf("Error encountered while storing Instancetype ControllerRevisions: %v", err), common.FailedCreateVirtualMachineReason), nil
+	// TODO(lyarwood): Extract all instance type logic into a handler
+	referencePolicy := c.clusterConfig.GetInstancetypeReferencePolicy()
```

Comment:
I would not go any further if the VM doesn't have an instance type or preference requested.
Could you move all of the code under an IF statement?

Reply:
I'll just extract the entire block and start with that conditional sure.

Reply:
https://github.com/kubevirt/kubevirt/pull/13086/commits/362391c55fcd34be4a3535cbdbe3f8cd772ff4e6